pandas/core/indexes/accessors.py

```diff
@@ -126,6 +126,48 @@ class DatetimeProperties(Properties):
     """
 
     def to_pydatetime(self):
+        """
+        Return an ndarray of native Python datetime objects.
+
+        Timezone information is retained if present.
+
+        .. warning::
+
+           Python's datetimes use microsecond resolution, which is lower than
+           pandas' (nanosecond). The values are truncated.
+
+        Returns
+        -------
+        numpy.ndarray
+            object dtype array containing native Python datetime objects.
+
+        See Also
+        --------
+        datetime.datetime : Standard library value for a datetime.
+
+        Examples
+        --------
+        >>> s = pd.Series(pd.date_range('20180310', periods=2))
+        >>> s.head()
+        0   2018-03-10
+        1   2018-03-11
+        dtype: datetime64[ns]
+
+        >>> s.dt.to_pydatetime()
+        array([datetime.datetime(2018, 3, 10, 0, 0),
+               datetime.datetime(2018, 3, 11, 0, 0)], dtype=object)
+
+        pandas' nanosecond precision is truncated to microseconds.
+
+        >>> idx = pd.date_range('2017', periods=2, freq='ns')
```

Comment:
can you use a Series example here?

Reply:
![nanosec](https://user-images.githubusercontent.com/21082240/37317675-8ad71562-2666-11e8-81aa-ea294aac6091.GIF)
Btw, I dont get the same output as you @TomAugspurger for the nanosecond-precision data

Reply:
Strange, not sure why that would be.

Are you able to wrap that in a Series and do `ser.dt.to_pydatetimte()`? I'll it once you update.